libsmtutil/CVC5Interface.cpp

```diff
@@ -0,0 +1,319 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libsmtutil/CVC5Interface.h>
+
+#include <libsolutil/CommonData.h>
+#include <libsolutil/CommonIO.h>
+#include <libsolutil/Exceptions.h>
+
+using namespace std;
+using namespace solidity;
+using namespace solidity::util;
+using namespace solidity::smtutil;
+
+CVC5Interface::CVC5Interface(optional<unsigned> _queryTimeout):
+	SolverInterface(_queryTimeout)
+{
+	reset();
+}
+
+void CVC5Interface::reset()
+{
+	m_variables.clear();
+	m_solver.resetAssertions();
+	m_solver.setOption("produce-models", "true");
+	if (m_queryTimeout)
+		m_solver.setOption("tlimit-per", std::to_string(*m_queryTimeout));
+	else
+		m_solver.setOption("rlimit", std::to_string(resourceLimit));
```

Comment:
It seems cvc5 has `tlimit` and `rlimit` and a global and per-query version of each: https://cvc5.github.io/docs/cvc5-1.0.2/resource-limits.html
Hopefully I got this correctly.